### PR TITLE
Printf annotation improvements

### PR DESCRIPTION
**Pull request by keszybz**
Fixes compilation under clang.

### CODE REVIEW

File: src/journal-remote/microhttpd-util.c [hunk 1]
```diff
@@ -103,7 +103,10 @@ int mhd_respondf(struct MHD_Connection *connection,
         errno = -error;
         fmt = strjoina(format, "\n");
```
Comment (martinpitt) [inline src/journal-remote/microhttpd-util.c:104]: The commit message is a copy&paste error, as in this case we are actually appending a known value (newline). But this is not fundamentally misleading, so no need (from my side) to repush just for this.

**Comment by martinpitt** [comment]
`FAIL: test_bash_crash` sounds a bit suspicious as this PR is related to logging and I don't remember seeing this fail before. i386 is still running, so I can't yet compare. I'll restart the amd64 test to make sure this doesn't regress.

**Comment by martinpitt** [comment]
Retried, and the [bash crash](https://anonscm.debian.org/cgit/pkg-systemd/systemd.git/tree/debian/tests/boot-and-services#n443) test consistently fails. This should be trivial to run manually, without invoking the full test suite. It runs

    ulimit -c unlimited; bash -c 'kill -SEGV $$'

and then waits for `/var/lib/systemd/coredump/core.bash.*` to appear (which works) and for this in the journal:

    systemd-coredump[1565]: Process 1563 (bash) of user 1000 dumped core.                                         
                                               Stack trace of thread 1563:

but this doesn't appear even after 10 seconds.

**Comment by poettering** [comment]
PR looks good to me otherwise.

**Comment by keszybz** [comment]
Oops, I really screwed this one up. There's an `if` under `ifdef` above there, which I forgot about. I reworked the patch with a different approach:
- expose log_dispatch
- use log_dispatch instead of log_struct

Seems to work fine here, both with journald and normal crashes.

**Pull request merged**